resources/assets/js/components/ApplicantProfile/Skills/List.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from "react";
+import { useIntl, defineMessages } from "react-intl";
+import {
+  Experience,
+  ExperienceSkill,
+  Skill,
+  SkillSkillCategory,
+  SkillCategory,
+} from "../../../models/types";
+import { getExperiencesOfSkill } from "../../Application/helpers";
+import { localizeFieldNonNull, getLocale } from "../../../helpers/localize";
+import { hasKey, sortLocalizedAlphabetical } from "../../../helpers/queries";
+import { localizedFieldNonNull } from "../../../models/app";
+import SkillAccordion from "./Accordion";
+
+const messages = defineMessages({
+  group: {
+    id: "applicantProfile.skills.list.group",
+    defaultMessage: "Group",
+    description: "Text for the 'Group' option on the sort toggle UI.",
+  },
+  alpha: {
+    id: "applicantProfile.skills.list.alpha",
+    defaultMessage: "A-Z",
+    description: "Text for the 'A-Z' option on the sort toggle UI.",
+  },
+  journey: {
+    id: "applicantProfile.skills.list.journey",
+    defaultMessage: "Journey",
+    description: "Text for the 'Journey' option on the sort toggle UI.",
+  },
+  sortBy: {
+    id: "applicantProfile.skills.list.sortBy",
+    defaultMessage: "Sort By:",
+    description: "Text for the label on the sort toggle UI.",
+  },
+});
+
+enum SortTypes {
+  Group = "group",
+  Alpha = "alpha",
+  // Journey = "journey", Future enhancement according to designs.
+}
+interface SortButtonProps {
+  text: string;
+  active: boolean;
+  handleClick: () => void;
+}
+
+const SortButton: React.FC<SortButtonProps> = ({
+  text,
+  active,
+  handleClick,
+}) => (
+  <button
+    data-h2-button={`${active ? "theme-1" : "black"}, square, medium, clear`}
+    type="button"
+    onClick={handleClick}
+  >
+    <span data-h2-button-label>{text}</span>
+  </button>
+);
+interface ListProps {
+  experiences: Experience[];
+  experienceSkills: ExperienceSkill[];
+  skillCategories: SkillCategory[];
+  skillSkillCategories: SkillSkillCategory[];
+  skills: Skill[];
+  applicantId: number;
+}
+
+const List: React.FC<ListProps> = ({
+  experiences,
+  experienceSkills,
+  skillCategories,
+  skillSkillCategories,
+  skills,
+  applicantId,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl.locale);
+  const [sortType, setSortType] = useState<SortTypes>(SortTypes.Group);
+
+  const skillCategoriesGroupedObject = skillCategories
```

Comment:
This looks like a lot of iterating, it might be worth wrapping this in `useMemo`.